kDrive/Utils/AppFactoryService.swift

```diff
@@ -20,19 +20,108 @@ import Foundation
 import InfomaniakDI
 import kDriveCore
 import os.log
+import UIKit
 
 /// Something that loads the DI on init
 public struct EarlyDIHook {
     public init(context: DriveAppContext) {
         os_log("EarlyDIHook")
 
-        let extraDependencies = [Factory(type: NavigationManageable.self) { _, _ in
-            NavigationManager()
-        }, Factory(type: AppContextServiceable.self) { _, _ in
-            AppContextService(context: context)
-        }]
+        let extraDependencies = [
+            Factory(type: NavigationManageable.self) { _, _ in
+                NavigationManager()
+            },
+            Factory(type: AppContextServiceable.self) { _, _ in
+                AppContextService(context: context)
+            },
+            Factory(type: AppRestorationService.self) { _, _ in
+                AppRestorationService()
+            }
+        ]
 
         // setup DI ASAP
         FactoryService.setupDependencyInjection(other: extraDependencies)
     }
 }
+
+// TODO: Refactor with Scenes / NSUserActivity
+#if !ISEXTENSION
+
+public final class AppRestorationService {
```

Comment:
move this code outside this file, probably requires some project changes but it will be cleaner.

Reply:
reworked a bit to tidy this up